Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift

```diff
@@ -253,11 +272,61 @@ extension PostCampaignCheckoutViewController: PledgeViewCTAContainerViewDelegate
 
   func submitButtonTapped() {
     self.paymentMethodsViewController.cancelModalPresentation(true)
-    // TODO: Respond to button tap
+    self.viewModel.inputs.submitButtonTapped()
   }
 
   func termsOfUseTapped(with helpType: HelpType) {
     self.paymentMethodsViewController.cancelModalPresentation(true)
     self.viewModel.inputs.termsOfUseTapped(with: helpType)
   }
 }
+
+// MARK: - PledgePaymentMethodsViewControllerDelegate
+
+extension PostCampaignCheckoutViewController: PledgePaymentMethodsViewControllerDelegate {
+  func pledgePaymentMethodsViewController(
+    _: PledgePaymentMethodsViewController,
+    didSelectCreditCard paymentSource: PaymentSourceSelected
+  ) {
+    switch paymentSource {
+    case let .paymentIntentClientSecret(clientSecret):
+      return STPAPIClient.shared.retrievePaymentIntent(withClientSecret: clientSecret) { intent, _ in
+        guard let intent = intent else {
+          self.messageBannerViewController?
+            .showBanner(with: .error, message: Strings.Something_went_wrong_please_try_again())
+          return
+        }
+
+        let stripeCardId = intent.stripeId
+        let paymentIntentClientSecret = paymentSource.paymentIntentClientSecret ?? ""
+
+        self.viewModel.inputs.creditCardSelected(with: (stripeCardId, paymentIntentClientSecret))
+      }
+    default:
+      break
```

Comment:
This also needs to handle the `.savedCreditCard` case - otherwise, we're not completing the pledge if you re-use an old card.

Reply:
Sheesh, thanks for the call-out! I got a little lost in the sauce with this one, and save cards slipped my mind. 

For saved cards I think we'll need to grab the stripeCardId on the selected stored card in the VM to pass to this mutation.

Reply:
Yeah, I think that's an argument for updating the enum to be 
```
case .savedCreditCard(String, String)
```